container/rkt/helpers.go

```diff
@@ -20,17 +20,56 @@ import (
 	"path"
 	"strings"
 
+	rktapi "github.com/coreos/rkt/api/v1alpha"
 	"github.com/golang/glog"
+	"golang.org/x/net/context"
 )
 
 type parsedName struct {
 	Pod       string
 	Container string
 }
 
-func verifyName(name string) (bool, error) {
-	_, err := parseName(name)
-	return err == nil, err
+func verifyPod(name string) (bool, bool, error) {
+	pod, err := cgroupToPod(name)
+
+	if err != nil || pod == nil {
```

Comment:
In `cgroupToPod` you return an error if the api didn't give you anything, which is that case.

The only time you could return a nil error and a nil pod is if the api returned an object of `{Pods: [nil]}` that it claims matches a name of whatever you passed in. That would be a significant enough bug that a panic would be fine imo (that is, it won't happen)